code/src/OSFramework/Grid/AbstractDataSource.ts

```diff
@@ -116,6 +118,22 @@ namespace OSFramework.Grid {
             this._isSingleEntity = false;
         }
 
+        private _getRowByKey(key: string) {
+            return this._ds.find((item) => {
```

Comment:
Can we get the row by getting the rownumber (with the getRowNumberByKey ) and then just use the rowNumber to get the row?